lib/krane/deploy_task.rb

```diff
@@ -1,12 +1,612 @@
 # frozen_string_literal: true
+require 'yaml'
+require 'shellwords'
+require 'tempfile'
+require 'fileutils'
 
-require 'kubernetes-deploy/deploy_task'
+require 'krane/common'
+require 'krane/concurrency'
+require 'krane/resource_cache'
+require 'krane/kubernetes_resource'
+%w(
+  custom_resource
+  cloudsql
+  config_map
+  deployment
+  ingress
+  persistent_volume_claim
+  pod
+  network_policy
+  service
+  pod_template
+  pod_disruption_budget
+  replica_set
+  service_account
+  daemon_set
+  resource_quota
+  stateful_set
+  cron_job
+  job
+  custom_resource_definition
+  horizontal_pod_autoscaler
+  secret
+).each do |subresource|
+  require "krane/kubernetes_resource/#{subresource}"
+end
+require 'krane/resource_watcher'
+require 'krane/kubectl'
+require 'krane/kubeclient_builder'
+require 'krane/ejson_secret_provisioner'
+require 'krane/renderer'
+require 'krane/cluster_resource_discovery'
+require 'krane/template_sets'
+require 'krane/deploy_task_config_validator'
 
 module Krane
-  class DeployTask < KubernetesDeploy::DeployTask
-    def initialize(**args)
-      raise "Use Krane::DeployGlobalTask to deploy global resources" if args[:allow_globals]
```

Comment:
This behavior is still important.

Reply:
I've done a poor job explaining this.

command                          | namespaced resources | global resources | both |
-------------------------|-------------------------|------------------|-----|
exe/kubernetes-deploy   | happy                               | prints a warning  | prints a warning |
krane deploy                     | happy                              | exception             | exception |
krane global-deploy         | exception                         | happy                   | exception |